packages/common/src/search/types/IHubSearchResult.ts

```diff
@@ -0,0 +1,105 @@
+import { IHubGeography } from "../..";
+import { AccessLevel, HubFamily, IHubEntityBase } from "../../core";
+
+/**
+ * Standardized light-weight search result structure, applicable to all
+ * types of search results - users, groups, content, events etc
+ */
+export interface IHubSearchResult extends IHubEntityBase {
+  /**
+   * Access level of the backing entity
+   */
+  access: AccessLevel;
+
+  /**
+   * Hub Family
+   */
+  family: HubFamily;
+
+  /**
+   * Owner; Applies to Items and Groups
+   */
+  owner: string;
+
+  /**
+   * Fully qualified thumbnail url for items, groups, users
+   * Will not have the token appended. Consuming app needs to
+   */
+  thumbnailUrl: string;
+  /**
+   * Hash of pre-computed urls
+   */
+  urls: {
+    /**
+     * Canononical ArcGIS Online Url for this entity (if applicable)
+     */
+    portalHome?: string;
+    /**
+     * Relative url, which can be appended to a Site's root Url
+     * e.g. /datasets/3ef8c7a, or /events/june-meeting
+     * This is much easier than passing a Site into the search
+     * system so the full url can be computed
+     */
+    relative: string;
```

Comment:
this seems like an odd attribute to have in the data model. the relative route is app-specific. If another developer is building an app they can use search results but have different routes.

Reply:
How we handle the urls is definitely a sticky aspect of this. Since there is (currently) a whole bunch of non-trivial logic in determining the url in the Hub, the idea was to encapsulate that so the client app can be ignorant of those details. We did setup the gallery to enable custom apps to provide a callback to compute the url differently, and I added the optional `customUrl` property to account for that.

Reply:
I suggest that instead of having the URL within the data object there could be a function like `hub.getRoute(entity):string` which would encapsulate (or delegate) the logic.

Reply:
Agreed to not add the token to the url